### PR TITLE
Migrating to Swift 5

### PR DESCRIPTION
**Pull request by jjorn**
Following updates have been done for Swift 5 compatibility:
- Migrate framework and demo project to Swift 5
- Upgrade 3rd-party frameworks
- Update badge in `README`

### CODE REVIEW

File: Demos/MungoHealer iOS-Demo/MungoHealer iOS-Demo/ViewController.swift [hunk 1]
```diff
@@ -13,16 +13,22 @@ struct MyLocalizedError: LocalizedError {
 }
 
 struct MyBaseError: BaseError {
+    var debugDescription: String?
```
Comment (Jeehut) [inline Demos/MungoHealer iOS-Demo/MungoHealer iOS-Demo/ViewController.swift]: Why was this added? Is this really necessary for Swift 5 support?

**Review by Jeehut — COMMENTED** [review]
Thanks for the PR. Looks good except my comment above. Please explain or fix.

File: Demos/MungoHealer iOS-Demo/MungoHealer iOS-Demo/ViewController.swift (hunk 1 above)
Comment (jjorn) [reply Demos/MungoHealer iOS-Demo/MungoHealer iOS-Demo/ViewController.swift]: When `debugDescription` is not explicitly defined, it triggers the error:

`Type 'MyBaseError' does not conform to protocol 'BaseError'`

Comment (Jeehut) [reply Demos/MungoHealer iOS-Demo/MungoHealer iOS-Demo/ViewController.swift]: This still doesn't clarify why adding the `debugDescription` too all types is the correct migration of the Swift 4.2 behavior to Swift 5. Which change in Swift did make this necessary? What is the recommended way to migrate from Swift 4.2 to Swift 5? These questions can only be answered if the rationale behind this change is clear. Therefore we should try to find the Swift Evolution prorposals which made this change to Swift. Only then it will be clear if and why this is the correct solution to the error.

Comment (jjorn) [reply Demos/MungoHealer iOS-Demo/MungoHealer iOS-Demo/ViewController.swift]: OK I've figure it out: what we need to do in this case, is just defining a default value with a proper access (e.g. `public`) at the protocol level. A default value with an appropriate access seems to be a must for an optional property.
